src/confd/src/ietf-interfaces.c

```diff
@@ -144,27 +144,33 @@ static int ifchange_cand_infer_vlan(sr_session_ctx_t *session, const char *path)
 			goto out_free_ifname;
 
 		*ptr++ = '\0';
-		vid = strtol(ptr, NULL, 10);
+		vid    = strtol(ptr, NULL, 10);
+		lower  = ifname;
 	} else if (!fnmatch("vlan+([0-9])", ifname, FNM_EXTMATCH)) {
 		if (sscanf(ifname, "vlan%d", &vid) != 1)
 		    goto out_free_ifname;
+
+		/* Avoid setting lower-layer-if to vlanN */
+		lower = NULL;
```

Comment:
What would be the harm of setting `lower = "br0";` here? In the common case where it exists, we save a few keystrokes; if not, then the user has to fill it in for themselves.

Reply:
Yeah I took the conservative approach from #531.  We should probably pull in @jovatn into this discussion, to me it's fine either way as long as we don't set `lower` to the same as `ifname` (the bug).

Reply:
I'll pick his brain on this AFK, thanks!

Now that I think about it, shouldn't there also be a `must` expression in the model that forbids setting the lower to itself? Otherwise we are still vulnerable to the user making the same mistake, right?

Reply:
Skipping.  AFK discussion decided current fix is the best one for now.